mobile/Ihc/components/MedicationInventory.js

```diff
@@ -4,69 +4,61 @@ import {
   Text,
   View
 } from 'react-native';
+
 import { Col, Row, Grid } from 'react-native-easy-grid';
 import UpdateMedicationModal from './UpdateMedicationModal';
+import {localData, serverData} from '../services/DataService';
 import Button from './Button';
-import Medication from '../models/Medication';
 
 export default class MedicationInventory extends Component<{}> {
   /*
    * Expects in props:
    *  {
    *    rows: [Medication],
-
-   *    saveModal: function
+   *    createMedication: function,
+   *    updateMedication: function
    *  }
    */
   constructor(props) {
     super(props);
     this.tableHeaders = ['Drug Name', 'Quantity', 'Dosage', 'Units', 'Notes'];
     this.rowNum = 0;
-
-    // showModal is the modal to update medication
-    // medicationToEdit is the medication that will be edited
-    this.state = {showModal: false, 
-      medicationToEdit: {
-        drugName: '',
-        quantity: 0,
-        dosage: 0,
-        units: '',
-        comments: '' //Consider keeping track of multiple comments (array of strings)
-      }
-    };
+    this.state = { showModal: false, oldKey: null, 
+      formValues: {drugName: null, quantity: null, dosage: null, units: null, comments: null}};
   }
 
-
-  openEditModal = (medicationToEdit) => {
-    this.setState({medicationToEdit: medicationToEdit, showModal: true});
+  openEditModal = (oldMedication) => {
+    const oldKey = oldMedication.key;
+    const formValues = this.getCurrentMedication(oldKey);
+    this.setState({ showModal: true, oldKey: oldKey, formValues: formValues});
   }
 
   openAddModal = () => {
-    this.setState({showModal: true, 
-      medicationToEdit: {
-        drugName: '',
-        quantity: 0,
-        dosage: 0,
-        units: '',
-        comments: '' 
-      }
-    });
+    this.setState({ showModal: true, oldKey: null, 
+      formValues: {drugName: null, quantity: null, dosage: null, units: null, comments: null}});
   }
 
   closeModal = () => {
-    this.setState({showModal: false, 
-      medicationToEdit: {
-        drugName: '',
-        quantity: 0,
-        dosage: 0,
-        units: '',
-        comments: '' 
-      }
-    });
+    this.setState({ showModal: false});
+  }
+  saveModal = (newMedication) => {
+    if (this.state.oldKey == null) {
+      this.props.createMedication(newMedication);
+    } else {
+      this.props.updateMedication(this.state.oldKey, newMedication);
+    }
   }
 
-  updateMedication = (newMedication) => {
-    this.setState({medicationToEdit: newMedication});
+  getCurrentMedication(oldKey) {
+    let medication = localData.getMedicationWithKey(oldKey);
+
+    // medication[0] since getMedicationWithKey gives
```

Comment:
I think itd make more sense to retrieve the element within the getMedWithKey method, because the name doesnt sound like it should return an array